golem-common/src/cache.rs

```diff
@@ -53,11 +52,10 @@ pub struct Cache<K, PV, V, E> {
     name: &'static str,
 }
 
-#[async_trait]
 pub trait SimpleCache<K, V, E> {
-    async fn get_or_insert_simple<F>(&self, key: &K, f: F) -> Result<V, E>
+    fn get_or_insert_simple<F>(&self, key: &K, f: F) -> impl Future<Output = Result<V, E>>
```

Comment:
Need this because I want to work with non-'static futures

Reply:
I see, need to look at your PR more where exactly we forced to make this change. It's interesting such a change was required.

Reply:
This guy will create a non-static future when used https://github.com/golemcloud/golem/pull/1444/files#diff-0b676bf48730736dc951c862a53c431e12ae2610bfbba134476906a6c4dbeddaR222 (which is fine, as we know exactly for how long it is needed).

Making it work with static futures is possible, but would mean cloning the Arc, introducing an Arc for the caches and introducing (and propagating) a Clone constraint for AuthCtx. I think this way is more elegant